### ISSUE TITLE
Some entries for macros missing in generated docs

### ISSUE THREAD

**Issue by magy00** [raspberrypi/pico-sdk#2129]
It looks like some macro entries like for `__not_in_flash(group)` or `__not_in_flash_func(func_name)` are missing in the online and PDF docs. Actually, their _example_ entries are there but under `__scratch_x(group)` and `__no_inline_not_in_flash_func(func_name)` instead:

https://www.raspberrypi.com/documentation/pico-sdk/runtime.html#macros-3
https://www.raspberrypi.com/documentation/pico-sdk/runtime.html#group_pico_platform_1gad8b78e068558946081789ce1935528bf
https://www.raspberrypi.com/documentation/pico-sdk/runtime.html#group_pico_platform_1gad9ab05c9a8f0ab455a5e11773d610787

**Comment by peternlewis** [comment]
I came here to report this as an issue. I'd try to figure out what the actual mistake is, but the doc generation is a complex system I can't really run myself.

My guess is that that unclosed `{` are the issue. and that the three lines like:

```
void __not_in_flash_func(my_func)(int some_arg) {
```

should be changed to

```
void __not_in_flash_func(my_func)(int some_arg);
```

**Comment by lurch** [comment]
> but the doc generation is a complex system I can't really run myself.

Have a look at Appendix B of https://pip.raspberrypi.com/documents/RP-009085-KB if you haven't already seen it.

**Comment by peternlewis** [comment]
Yes, I refer to that document regularly, it is excellent, which is why I am reporting the minor error in the handling of that specific area. Indeed in the PDF linked, it has the same issue (see section 5.5.16.4.13) - it is a mishmash of `__not_in_flash_func`,  `_no_inline_not_in_flash_func`, and `__time_critical_func`, all in the `5.5.16.4.13. __no_inline_not_in_flash_func` section because of the mis-parsing of the header file.

My point about the doc generation was simply that I am unable to generate the docs myself and thus try to figure out what in particular is causing the error in parsing, but my guess is it is the unmatched `{` as I could not see anything else in there that looked clearly different to other sections.
